### ISSUE TITLE
Performance bottleneck in package-symbols

### ISSUE THREAD

**Issue by Ambrevar** [atlas-engineer/nsymbols#3]
In my experience package-symbols got to be superfast.  When visibility is about all symbols and all packages, it quickly gets slow.

For instance, the `(if (eq :external visibility)` could be moved out of the loop which could increase performance.

In any case, having a benchmark test would be useful to track regressions.

**Comment by aartaka** [comment]
Yes, it's quite slow. Moving the condition out of the loop made it even slower for me, though :D

**Comment by Ambrevar** [comment]
That's why we need benchmarks :p
